examples/with-apollo/lib/apollo.js

```diff
@@ -92,68 +141,31 @@ export const withApollo = ({ ssr = true } = {}) => PageComponent => {
               props = { pageProps: { ...pageProps, apolloClient } }
             }
 
-            // Takes React AppTree, determine which queries are needed to render,
-            // then fetche them all.
+            // Take the Next.js AppTree, determine which queries are needed to render,
+            // and fetch them. This method can be pretty slow since it renders
+            // your entire AppTree once for every query. Check out apollo fragments
+            // if you want to reduce the number of rerenders.
+            // https://www.apollographql.com/docs/react/data/fragments/
             await getDataFromTree(<AppTree {...props} />)
           } catch (error) {
             // Prevent Apollo Client GraphQL errors from crashing SSR.
             // Handle them in components via the data.error prop:
             // https://www.apollographql.com/docs/react/api/react-apollo.html#graphql-query-data-error
             console.error('Error while running `getDataFromTree`', error)
           }
-
-          // getDataFromTree does not call componentWillUnmount
-          // head side effect therefore need to be cleared manually
-          Head.rewind()
```

Comment:
According to @timneutkens , this should not have been removed: https://github.com/zeit/next.js/issues/9326#issuecomment-590740693

Reply:
Thanks for clarifying 🙏. https://github.com/zeit/next.js/pull/10696